### PR TITLE
secrets: In-file key management

### PR DESCRIPTION
**Pull request by vain**

### CODE REVIEW

File: bundlewrap/secrets.py [hunk 1]
```diff
@@ -97,35 +97,48 @@ def _decrypt(self, cryptotext=None, key='encrypt'):
...
+        content = get_file_contents(join(self.repo.data_dir, source_path))
+
+        key_delim = content.find(b'$')
+        if key_delim > -1:
+            key_from_file = content[:key_delim].decode('utf-8')
+            content = content[key_delim + 1:]
+        else:
+            key_from_file = None
+
+        if key is None:
```
Comment (vain) [inline bundlewrap/secrets.py]: A key given as a parameter to this function wins over a key from the file. Why? Suppose decrypting the file using the key from the file fails. Now you want to test if another key works, so you use `bw debug -c "print(repo.vault.decrypt_file('test', key='hurz'))"`.

**Comment by trehn** [comment]
Can you add a line to https://docs.bundlewrap.org/guide/secrets/#key-management explaining this?

File: docs/content/guide/secrets.md [hunk 2]
```diff
@@ -103,10 +103,12 @@ If the source file is binary however (or any encoding other than UTF-8), you mus
 
 You can always add more keys to your `.secrets.cfg`, but you should keep the defaults around. Adding more keys makes it possible to give different keys to different teams. **By default, BundleWrap will skip items it can't find the required keys for**.
 
-When using `.password_for()`, `.decrypt()` etc., you can provide a `key` argument to select the key:
+When using `.password_for()`, `.encrypt()` etc., you can provide a `key` argument to select the key:
 
 	repo.vault.password_for("some database", key="devops")
 
+The encrypted data will be prefixed by `yourkeyname$...` to indicate that the key `yourkeyname` was used for encryption. Thus, during decryption, you can emit the `key=` parameter.
```
Comment (trehn) [inline docs/content/guide/secrets.md]: s/emit/omit/

**Comment by CroneKorkN** [comment]
\o/

**Pull request merged**